tests/phpunit/api/v3/JobTest.php

```diff
@@ -882,6 +882,30 @@ public function testBatchMergeIDRange() {
 
   }
 
+  /**
+   * Test the batch merge copes with custom data in base config.
+   *
+   * The custom data may not be copied if customTree returns nothing - odd!
+   */
+  public function testBatchMergeCustomData() {
+    CRM_Core_Config::singleton()->userPermissionClass->permissions = array('access CiviCRM', 'edit my contact');
+    $mouseParams = ['first_name' => 'Mickey', 'last_name' => 'Mouse', 'email' => 'tha_mouse@mouse.com'];
+    $this->individualCreate($mouseParams);
+
+    $customGroup = $this->CustomGroupCreate();
+    $customField = $this->customFieldCreate(array('custom_group_id' => $customGroup['id']));
```

Comment:
Does this create a view only field?

Reply:
it's at the field level

Reply:
so that line is creating the CustomField but i'm not seeing that its setting the custom Field to be view only?

Reply:
You are right - it seems to have gotten lost in some variant of this - have pushed it now - OK for merge on pass?

(I want to put the 4.7.31 rc onto staging for rc testing)